modules/ingester/instance.go

```diff
@@ -154,14 +144,39 @@ func newInstance(instanceID string, limiter *Limiter, overrides ingesterOverride
 	return i, nil
 }
 
-func (i *instance) PushBytesRequest(ctx context.Context, req *tempopb.PushBytesRequest) error {
+func (i *instance) PushBytesRequest(ctx context.Context, req *tempopb.PushBytesRequest) *tempopb.PushResponse {
+	atLeastOneError := false
+	pr := &tempopb.PushResponse{}
+
 	for j := range req.Traces {
+
 		err := i.PushBytes(ctx, req.Ids[j].Slice, req.Traces[j].Slice)
 		if err != nil {
-			return err
+			// only make list if there is at least one error
+			if !atLeastOneError {
+				pr.ErrorsByTrace = make([]tempopb.PushErrorReason, 0, len(req.Traces))
+				// because this is the first error, fill list with NO_ERROR for the traces before this one
+				for k := 0; k < j; k++ {
+					pr.ErrorsByTrace = append(pr.ErrorsByTrace, tempopb.PushErrorReason_NO_ERROR)
+				}
+
+				atLeastOneError = true
+			}
+			if errors.Is(err, errMaxLiveTraces) {
+				pr.ErrorsByTrace = append(pr.ErrorsByTrace, tempopb.PushErrorReason_MAX_LIVE_TRACES)
+				continue
+			}
+
+			if errors.Is(err, errTraceTooLarge) {
+				pr.ErrorsByTrace = append(pr.ErrorsByTrace, tempopb.PushErrorReason_TRACE_TOO_LARGE)
+				continue
+			}
```

Comment:
if error is neither max live traces or trace too large do we just swallow it here? should we return other errors?